src/plone/restapi/serializer/dxcontent.py

```diff
@@ -138,8 +139,23 @@ def __call__(self, version=None):
         if batch.links:
             result['batching'] = batch.links
 
-        result['items'] = [
-            getMultiAdapter((brain, self.request), ISerializeToJsonSummary)()
-            for brain in batch
-        ]
+        if 'fullobjects' in self.request.form.keys():
+            handler = SearchHandler(self.context, self.request)
+            result['items'] = handler.search(
+                query={
+                    'fullobjects': 'true',
+                    'path': {
+                        'query': '/'.join(self.context.getPhysicalPath()),
+                        'depth': 1
+                    }
+                }).get('items')
+        else:
+            result['items'] = [
+                getMultiAdapter(
+                    (brain, self.request),
+                    ISerializeToJsonSummary
+                )()
+                for brain in batch
+            ]
+
```

Comment:
I think that these two codepaths can be merged into one with something like
```python
query = {'path': {
    'query': '/'.join(self.context.getPhysicalPath()),
    'depth': 1
}
if 'fullobjects' in self.request.form:
    query['fullobjects'] = True

# Contains batching from withing the Lazy serializer
result = handler.search(query)
return results
```

Reply:
Right, so one path uses ISerializeToJsonSummary and the other uses ISerializeToJson.